### PR TITLE
Reduce unit test latency: only one etcd registry prefix, api version; do not duplicate builds

### PR DESCRIPTION
**Pull request by fejta**
Ref #25940

WIll just test one on presubmit; will change postcommit jenkins to do everything
Also skip hack/build-go.sh as this is already done before the tests start.

### CODE REVIEW

**Comment by ixdy** [comment]
Maybe remove the `./hack/build-go.sh` line too?

**Comment by fejta** [comment]
@ixdy removed that line

File: hack/jenkins/test-dockerized.sh [hunk 1]
```diff
@@ -36,23 +36,20 @@ export PATH=${GOPATH}/bin:${PWD}/third_party/etcd:/usr/local/go/bin:${PATH}
...
 export ARTIFACTS_DIR=${WORKSPACE}/artifacts
-# Save the verbose stdout as well.
-export KUBE_KEEP_VERBOSE_TEST_OUTPUT=y
-export KUBE_TIMEOUT='-timeout 300s'
+export KUBE_COVER="n"  # Disable coverage report
 export KUBE_INTEGRATION_TEST_MAX_CONCURRENCY=4
+export KUBE_JUNIT_REPORT_DIR=${WORKSPACE}/artifacts  # junix xml files
+export KUBE_KEEP_VERBOSE_TEST_OUTPUT=y  # Save the verbose stdout as well.
+export KUBE_RACE=-race  # Enable the Go race detector.
+export KUBE_TEST_API_VERSIONS="${KUBE_TEST_API_VERSIONS:-v1}"
```
Comment (ixdy) [inline hack/jenkins/test-dockerized.sh]: I think we should maybe only override these for the PR Jenkins job.

Comment (ixdy) [inline hack/jenkins/test-dockerized.sh]: https://github.com/kubernetes/test-infra/pull/80#discussion_r64849778

Comment (fejta) [inline hack/jenkins/test-dockerized.sh]: The PR jenkins job isn't source controlled. This is so we need to do it here.

Comment (ixdy) [inline hack/jenkins/test-dockerized.sh]: if you feel this strongly about it being in source control in this file, please
* open an issue to revert this change and refer to it in a comment
* update `hack/jenkins/gotest-dockerized.sh` to pass in the Jenkins job name environment variable
* guard overriding the defaults so that it only occurs on the PR Jenkins unit/integration job

**Comment by ixdy** [comment]
I guess we can't get rid of `hack/build-go.sh` entirely:
```
+++ [0526 20:12:31] Running kubectl with no options
./hack/test-cmd.sh: line 152: /go/src/k8s.io/kubernetes/_output/local/bin/linux/amd64/kubectl: No such file or directory
!!! Error in ./hack/test-cmd.sh:152
  '"${KUBE_OUTPUT_HOSTBIN}/kubectl"' exited with status 1
```
`hack/test-cmd.sh` should probably ensure that necessary binaries are built. that's a more invasive change, though.
